platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventUnhashedData.java

```diff
@@ -31,16 +31,36 @@
  * A class used to store base event data that does not affect the hash of that event.
  * <p>
  * A base event is a set of data describing an event at the point when it is created, before it is added to the
- * hashgraph and before its consensus can be determined. Some of this data is used to create a hash of an event
- * that is signed, and some data is additional and does not affect that hash. This data is split into 2 classes:
+ * hashgraph and before its consensus can be determined. Some of this data is used to create a hash of an event that is
+ * signed, and some data is additional and does not affect that hash. This data is split into 2 classes:
  * {@link BaseEventHashedData} and {@link BaseEventUnhashedData}.
  */
 public class BaseEventUnhashedData implements SelfSerializable {
     private static final long CLASS_ID = 0x33cb9d4ae38c9e91L;
-    private static final int CLASS_VERSION = 1;
     private static final int MAX_SIG_LENGTH = 384;
     private static final long SEQUENCE_UNUSED = -1;
 
+    private static class ClassVersion {
+        /**
+         * The original version of the BaseEventUnhashedData class.
+         */
+        public static final int ORIGINAL = 1;
+
+        /**
+         * Removes the serialization of the sequence information and other parent creator id.
+         *
+         * @since 0.45.0
+         */
+        public static final int ADDRESS_BOOK_ROUND = 2;
+    }
+
+    /**
+     * The version of the software discovered during deserialization that needs to be preserved in serialization.
+     * <p>
+     * DEPRECATED: Remove after 0.45 is delivered to mainnet.
+     */
+    private int deserializedVersion = ClassVersion.ADDRESS_BOOK_ROUND;
```

Comment:
Why is it needed? Usually the idea is to serialize according to the "current" (latest) version, while during deserialization support a few previous versions with default values for the newly added fields. I mean, if an object is deserialized from an old version, it will have some fields null / default, but when it needs to be serialized again, there is no need to serialize it in the old format

Reply:
Supporting the prior serialization scheme is needed due to these factors: 
*  NEW: At startup, events are replayed from the saved pre-consensus event stream which are in the older serialization scheme
*  These replayed events must go through validation, including signature validation
*  This validation requires hashing the events and obtaining the exact same signature
*  Hashing uses the serialized format, so the events must be serializable in the older format to derive the saved signature.

Not doing the above leads to ISS. 

This is only relevant when we make a change to the serialization scheme, that we have to support the older serialization scheme until all events are serialized in the new serialization scheme. 

The code is configured to only serialize in the older serialization scheme if the data was deserialized from the older serialization scheme.   All new events will be serialized in the new serialization scheme.